### ISSUE TITLE
false negative await-async-query error on a similar-named method from another library

### ISSUE THREAD

**Issue by Jalilvand-MJ** [testing-library/eslint-plugin-testing-library#613]
### Have you read the Troubleshooting section?

Yes

### Plugin version

5.5.0

### ESLint version

8.2.0

### Node.js version

14.20

### package manager and version

npm 8.14

### Operating system

Ubuntu 20.04.4

### Bug description

In tests I've mocked G6 Graph object and called `findById` method on the instance, which returns an object but not a promise.
Because of having same name with testing library's async query, the rule reports error of not handled promise.

### Steps to reproduce

1. Write an arbitrary, not-async function, with name `findById`, and with any return type but Promise
2. Call it in the test
3. Run eslint

### Error output/screenshots

ESLint: promise returned from `findById` query must be handled(testing-library/await-async-query)

### ESLint configuration

```
  "extends": [
    ...
    "plugin:testing-library/react",
    ...
  ],
```

### Rule(s) affected

testing-library/await-async-query

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
This false positive is related to the Aggressive Reporting feature. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by Jalilvand-MJ** [comment]
Thanks for your help
